client/hotrod-client/src/main/java/org/infinispan/client/hotrod/TransactionalRemoteCache.java

```diff
@@ -0,0 +1,23 @@
+package org.infinispan.client.hotrod;
+
+import javax.transaction.Transaction;
+import javax.transaction.TransactionManager;
+
+/**
+ * A {@link RemoteCache} implementation that handles {@link Transaction}.
+ * <p>
+ * All streaming operation (example {@link TransactionalRemoteCache#retrieveEntries(String, int)}) and {@link
+ * TransactionalRemoteCache#size()} aren't transactional in a way they don't interact with the transaction's data (keys,
+ * values).
+ * <p>
+ * {@link TransactionalRemoteCache#containsValue(Object)} is a special case where a key with the specific value is
+ * marked as read fpr the transaction.
+ *
+ * @author Pedro Ruivo
+ * @since 9.3
+ */
+public interface TransactionalRemoteCache<K, V> extends RemoteCache<K, V> {
```

Comment:
I don't like the asymmetry this creates with embedded caches, where getTransactionManager() is a member of AdvancedCache.
Why not create a common TransactionalCache interface with the method and have both AdvancedCache and TransactionalRemoteCache implement it.
Additionally, maybe we should take the opportunity to return introduce an AsyncTransactionManager where commit and rollback return CompletableFutures